### ISSUE TITLE
`prefer-const` should consider block scope

### ISSUE THREAD

**Issue by timmolendijk** [eslint/eslint#4838]
The documentation for `prefer-const` mentions the following pattern as a problem:

```js
// `a` is re-defined (not modified) on each loop step.
for (let a of [1,2,3]) { /*error `a` is never modified, use `const` instead.*/
    console.log(a);
}
```

But using `const` here would **not work** as desired:
```js
for (const a of [1,2,3]) {
    console.log(a);
}
// logs:
// 1
// 1
// 1
```

(ESLint v1.10.3 with `"prefer-const": 1`)

**Comment by eslintbot** [comment]
@timmolendijk Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by michaelficarra** [comment]
You are incorrect, const should work fine in correct implementations. Please search for duplicates before opening an issue next time.

**Comment by timmolendijk** [comment]
Hmm, yes you are right. I tested it without `'use strict';` - I guess that's where I went wrong.

Btw, I did search for a similar issue before posting but couldn't find one! Sorry anyway. And thanks.

**Comment by michaelficarra** [comment]
Yeah, GitHub search is bad, but I figured something would have shown up given the number of times this has come up. A thousand curses upon the V8 developers for making us have to deal with this nonsense.
